econml/inference/_inference.py

```diff
@@ -538,11 +536,11 @@ class InferenceResults(metaclass=abc.ABCMeta):
         The transform function to get the corresponding feature names from featurizer
     """
 
-    def __init__(self, d_t, d_y, pred, inf_type, fname_transformer=lambda nm: nm,
+    def __init__(self, d_t, d_y, pred, inf_type, fname_transformer=None,
                  feature_names=None, output_names=None, treatment_names=None):
         self.d_t = d_t
         self.d_y = d_y
-        self.pred = pred
+        self.pred = copy.deepcopy(pred)
```

Comment:
Since `pred` is array-like, I'd use

```suggestion
        self.pred = np.copy(pred)
```

instead; this is a shallow copy, but may be more efficient for array data (and we shouldn't need a deep copy for this data anyway).

Reply:
As we discussed, we don't even need to copy attribute here in the way we call InferenceResults, I will revert those changes and reply to that issue.